AT-3/src/at3/Vegetable.java

```diff
@@ -0,0 +1,88 @@
+package at3;
+
+import java.util.Comparator;
+
+public class Vegetable {
+    private String name;
+    private String color;
+    private double weight;
+    private double energyDensity; //kcal/100g
+
+    Vegetable(String name, String color, double weight, double energyDensity) {
+        this.name = name;
+        this.color = color;
+        this.weight = weight;
+        this.energyDensity = energyDensity;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public double getWeight() {
+        return weight;
+    }
+
+    public double getEnergyDensity() {
+        return energyDensity;
+    }
+
+    public double calcCalories() {
+        double result = (energyDensity * weight) / 100;
+        return result;
+    }
+    public boolean equals(Vegetable v) {
+        if(this == v || (this.getWeight() == v.getWeight()) && this.getName() == v.getName() && this.getColor() == v.getColor())  {

```

Comment:
this == v what you would like to achieve here? 
Example:
"hello" == "hello"
will equate to true, but
new String("hello") == new String("hello")
equates to false. 
This happens because in the second case, you created two new objects, whereas in the first case, the two String literals both point to the same object in the String pool.

Reply:
in this case I cannot see implementation of hashCode overriding, to make it work as expected

Reply:
this == v - checks if it's this and v are the same object